graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +57,262 @@ def get_directives_for_ast(ast):
+            name_to_use = min(tag_names, key=len)
+        elif len(user_specified_names) == 1:
+            name_to_use = next(iter(user_specified_names))
+        else:
+            raise AssertionError(u'Multiple tags on the same field are not allowed: {}'
+                                 .format(user_specified_names))
+
+        # Remove all tags but the one we decided to keep
+        name_change_map = {
+            name: name_to_use
+            for name in tag_names
+        }
+        new_directives = [
+            directive
+            for directive in ast.directives
+            if (directive.name.value != TagDirective.name or
+                directive.arguments[0].value.value == name_to_use)
+        ]
+
+    return made_changes, new_directives, name_change_map
+
+
+def _remove_colocated_tags(non_macro_names, ast):
+    """Return an AST with at most one tag per field by removing tags.
+
+    See merge_colocated_tags in this module for an example usage.
+
+    Args:
+        non_macro_names: set of tag names that the user wrote. We prefer keeping these names.
+        ast: GraphQL query AST object that potentially has multiple colocated tags. This AST
+             should not contain any duplicate tags (different tags with the same name).
+
+    Returns:
+        tuple(new_ast, name_change_map). new_ast is the ast with at most one tag per field.
+        name_change_map is a dict (string -> string) that contains the new name for each
+        tag name. Names of removed tags are mapped to the name of the colocated tag that was
+        not removed.
+    """
+    if not isinstance(ast, (Field, InlineFragment, OperationDefinition)):
+        return ast
+
+    made_changes = False
+    name_change_map = dict()
+
+    # Recurse
+    new_selection_set = None
+    if ast.selection_set is not None:
+        new_selections = []
+        for selection_ast in ast.selection_set.selections:
+            new_selection_ast, inner_name_change_map = _remove_colocated_tags(
+                non_macro_names, selection_ast)
+
+            name_collisions = set(name_change_map.keys()) & set(inner_name_change_map.keys())
+            if name_collisions:
+                raise AssertionError(u'The ast should not contain multiple tags with '
+                                     u'the same name. Found duplicate on names: {}'
+                                     .format(name_collisions))
+            name_change_map.update(inner_name_change_map)
+
+            if selection_ast is not new_selection_ast:
+                # Since we did not get the exact same object as the input, changes were made.
+                # That means this call will also need to make changes and return a new object.
+                made_changes = True
+
+            new_selections.append(new_selection_ast)
+        new_selection_set = SelectionSet(new_selections)
+
+    # Process the current node
+    made_changes_at_this_node, new_directives, new_name_change_map = \
```

Comment:
nit: use parethensis over \

Reply:
The function name doesn't fit on the first line. Are you suggesting splitting the lvalue in two lines?